Replace assert with cuda assert macro

serhaty opened the pull request:
Summary:
Follow-up to https://github.com/pytorch/pytorch/pull/31276

This is final replacement needed for aten out of place hipification.

Test Plan: wait for CI to clear.

Reviewed By: bddppq

Differential Revision: D19070209

kostmo commented:
## CircleCI build failures summary
As of commit b9cd3ff:
* **1/1** broken upstream at merge base 36d17f4 (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=36d17f4105f48d264fe085ce3dc0e4e825ebfb42&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
          ```
          git fetch viable/strict
          git rebase --onto viable/strict $(git merge-base origin/master HEAD)
          ```
        * If your commit is older than `viable/strict`:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
* **0/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=b9cd3ff6d1896687bd0bd2d901930857e66e4826).

### 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably caused by upstream breakages:
* [pytorch_linux_xenial_cuda9_cudnn7_py3_build](https://dr.pytorch.org/build-details.html?build_id=74507038)



---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

Commit:
Replace assert with cuda assert macro (#31297)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/31297

Follow-up to https://github.com/pytorch/pytorch/pull/31276

This is final replacement needed for aten out of place hipification.

Test Plan: wait for CI to clear.

Reviewed By: bddppq

Differential Revision: D19070209

fbshipit-source-id: 8c8f32e7bd23c9f5b534e1523ac7281c6d3418bf
